controllers/flinkcluster_submit_job_script.go

```diff
@@ -66,19 +66,29 @@ function wait_for_job() {
 	while true; do
 		echo -e "\nWaiting for job to finish..."
 		list_jobs
-		if list_jobs | grep "(FINISHED)"; then
-			echo -e "\nJob has completed successfully, exiting 0"
-			return 0
-		fi
-		if list_jobs | grep "(FAILED)"; then
-			echo -e "\nJob failed, exiting 1"
-			return 1
-		fi
-		if list_jobs | grep "(CANCELED)"; then
-			echo -e "\nJob has been cancelled, exiting 2"
-			return 2
+
+		# Find active job first.
+		# If the current job is restarted by the operator, there will be records of past stopped jobs.
+		# TODO: It needs to be improved to determine the job state with the submitted job id.
+		if list_jobs | grep -e "(SCHEDULED)" -e "(RUNNING)"; then
+			echo -e "\nFound an active job."
+			sleep 30
```

Comment:
I prefer putting `sleep 30` to outside of if-else, because if none of of these states are matched, we don't get into a busy wait situation.

Reply:
It looks better. I've fixed.